File: CodingDojo/games/reversi/src/main/java/com/codenjoy/dojo/reversi/model/Elements.java
package com.codenjoy.dojo.reversi.model;

/*-
 * #%L
 * Codenjoy - it's a dojo-like platform from developers to developers.
 * %%
 * Copyright (C) 2018 Codenjoy
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */


import com.codenjoy.dojo.services.printer.CharElements;

public enum Elements implements CharElements {

    NONE(' '),          // пустое место для фишки
    BREAK('☼'),         // препятствие на котором ставить фишку нельзя

    WHITE('o'),         // белые фишки, ходят черные
    WHITE_STOP('.'),    // белые фишки, не мой (белых) ход сейчас
    WHITE_TURN('O'),    // белые фишки, ходят белые

    BLACK('x'),         // черные фишки, ходят белые
    BLACK_STOP('+'),    // черные фишки, не мой (черных) ход сейчас
    BLACK_TURN('X');    // черные фишки, ходят черные

    final char ch;

    Elements(char ch) {
        this.ch = ch;
    }

    @Override
    public char ch() {
        return ch;
    }

    @Override
    public String toString() {
        return String.valueOf(ch);
    }

    public static Elements valueOf(char ch) {
        for (Elements el : Elements.values()) {
            if (el.ch == ch) {
                return el;
            }
        }
        throw new IllegalArgumentException("No such element for " + ch);
    }

}

File: CodingDojo/games/spacerace/src/main/python/elements.py
#! /usr/bin/env python3

###
# #%L
# Codenjoy - it's a dojo-like platform from developers to developers.
# %%
# Copyright (C) 2018 Codenjoy
# %%
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public
# License along with this program.  If not, see
# <http://www.gnu.org/licenses/gpl-3.0.html>.
# #L%
###
from enum import Enum

class Elements(Enum):
    NONE        = (0, ' ')
    BULLET_PACK = (1, '7')
    HERO        = (2, '☺')
    OTHER_HERO  = (3, '☻')
    DEAD_HERO   = (4, '+')
    STONE       = (5, '0')
    BOMB        = (6, '♣')
    EXPLOSION   = (7, 'x')
    BULLET      = (8, '*')
    GOLD        = (9, '$')
    WALL        = (10,'☼')

    def __init__(self, id: int, char: str):
        self.id = id
        self.char = char

    def __hash__(self):
        return hash(self.char)
        
    def getByChar(char):
        return __Elements__[char]


__Elements__ = dict(map(lambda element : (element.char, element), Elements))

if __name__ == '__main__':
    raise RuntimeError("This module is not designed to be ran from CLI")